icicle-source/src/Icicle/Source/Transform/ReifyPossibility.hs

```diff
@@ -275,7 +275,16 @@ makeApps a fun (arg:rest) doWrap
             -- Bare value. Note that this is now definite, but with same (bare) type
             bare  = Var (extractValueAnnot arga) nValue
 
-        fun' <- makeApps a (App a fun bare) rest True
+        -- Check if we're calling a primitive which already returns its result wrapped.
+        -- If so, we do not need to perform rewrapping of the value.
+        let doWrap'
+              | Just (p, _, _) <- takePrimApps fun
+              , primReturnsPossibly p
```

Comment:
"box" does, but it's desugared to a simple case expression in reifyPossibilityX: https://github.com/ambiata/icicle/blob/b3884e9f6fb84550aaad3c3d6c474d695ed12185/icicle-source/src/Icicle/Source/Transform/ReifyPossibility.hs#L47
It's harder to do the same here because then we'd kind of need two primitives, one that does the check and one that doesn't. Then we'd desugar the 'checked' one to the 'unsafe' one.